Code/Framework/AzCore/AzCore/Metrics/JsonTraceEventLogger.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Metrics/IEventLogger.h>
+
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/std/containers/fixed_vector.h>
+#include <AzCore/std/containers/ring_buffer.h>
+#include <AzCore/std/functional.h>
+#include <AzCore/std/parallel/mutex.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+
+namespace AZ::IO
+{
+    class GenericStream;
+}
+
+namespace AZ::Metrics
+{
+    // Contains JsonTraceEventLogger specific configuration
+    struct JsonTraceLoggerEventConfig
+    {
+        AZStd::string_view m_loggerName;
+    };
+
+    class JsonTraceEventLogger
+        : public IEventLogger
+    {
+    public:
+        JsonTraceEventLogger();
+        explicit JsonTraceEventLogger(JsonTraceLoggerEventConfig);
+        //! Generic stream which owned by the JsonEventLogger
+        explicit JsonTraceEventLogger(AZStd::unique_ptr<AZ::IO::GenericStream> stream);
+        JsonTraceEventLogger(AZStd::unique_ptr<AZ::IO::GenericStream> stream, JsonTraceLoggerEventConfig);
+
+        // Json Trace Event Logger is movable, but not copyable
```

Comment:
This comment looks like it belongs to a delete of the copy constructor, not the destructor?

Reply:
This comment is actually incorrect.
It is neither copyable nor movable.
I'll delete it